core/templates/dev/head/pages/exploration_player/ExplorationPlayerStateService.ts

```diff
@@ -147,11 +181,22 @@ oppia.factory('ExplorationPlayerStateService', [
           initExplorationPlayer(callback);
         }
       },
+      initializeQuestionPlayer: function(config, callback) {
```

Comment:
is there any specific reason for using `ExplorationPlayerStateService` for questionPlayer? I don't see any use of existing functionality of this file in the new `initializeQuestionPlayer` function. I think we should create a separate service for handling question-related stuff.)

Reply:
The question player is also a part of the exploration because in the future pretests and review tests (which are a part of the exploration) will be using the question player. So it needs to be part of that. All of the functionality is common in ConversationSkinDirective (which is common for both players). The only difference is the new method I added where we decide what data to fetch, hope that makes sense!